terraform/graph_config_node_module.go

```diff
@@ -0,0 +1,230 @@
+		t := &ModuleDestroyTransformer{}
+		if err := t.Transform(graph); err != nil {
+			return nil, err
+		}
+	}
+
+	// Build the actual subgraph node
+	return &graphNodeModuleExpanded{
+		Original:  n,
+		Graph:     graph,
+		Variables: t.Variables,
+	}, nil
+}
+
+// GraphNodeExpandable
+func (n *GraphNodeConfigModule) ProvidedBy() []string {
+	// Build up the list of providers by simply going over our configuration
+	// to find the providers that are configured there as well as the
+	// providers that the resources use.
+	config := n.Tree.Config()
+	providers := make(map[string]struct{})
+	for _, p := range config.ProviderConfigs {
+		providers[p.Name] = struct{}{}
+	}
+	for _, r := range config.Resources {
+		providers[resourceProvider(r.Type, r.Provider)] = struct{}{}
+	}
+
+	// Turn the map into a string. This makes sure that the list is
+	// de-dupped since we could be going over potentially many resources.
+	result := make([]string, 0, len(providers))
+	for p, _ := range providers {
+		result = append(result, p)
+	}
+
+	return result
+}
+
+// graphNodeModuleExpanded represents a module where the graph has
+// been expanded. It stores the graph of the module as well as a reference
+// to the map of variables.
+type graphNodeModuleExpanded struct {
+	Original *GraphNodeConfigModule
+	Graph    *Graph
+
+	// Variables is a map of the input variables. This reference should
+	// be shared with ModuleInputTransformer in order to create a connection
+	// where the variables are set properly.
+	Variables map[string]string
+}
+
+func (n *graphNodeModuleExpanded) Name() string {
+	return fmt.Sprintf("%s (expanded)", dag.VertexName(n.Original))
+}
+
+func (n *graphNodeModuleExpanded) ConfigType() GraphNodeConfigType {
+	return GraphNodeConfigTypeModule
+}
+
+// GraphNodeDependable
+func (n *graphNodeModuleExpanded) DependableName() []string {
+	return n.Original.DependableName()
+}
+
+// GraphNodeDependent
+func (n *graphNodeModuleExpanded) DependentOn() []string {
+	return n.Original.DependentOn()
+}
+
+// GraphNodeDotter impl.
+func (n *graphNodeModuleExpanded) DotNode(name string, opts *GraphDotOpts) *dot.Node {
+	return dot.NewNode(name, map[string]string{
+		"label": dag.VertexName(n.Original),
+		"shape": "component",
+	})
+}
+
+// GraphNodeEvalable impl.
+func (n *graphNodeModuleExpanded) EvalTree() EvalNode {
+	var resourceConfig *ResourceConfig
+	return &EvalSequence{
+		Nodes: []EvalNode{
+			&EvalInterpolate{
+				Config: n.Original.Module.RawConfig,
+				Output: &resourceConfig,
+			},
+
+			&EvalVariableBlock{
+				Config:    &resourceConfig,
+				Variables: n.Variables,
+			},
+		},
+	}
+}
+
+// GraphNodeFlattenable impl.
+func (n *graphNodeModuleExpanded) FlattenGraph() *Graph {
+	graph := n.Subgraph()
+	input := n.Original.Module.RawConfig
+
+	// Go over each vertex in the graph and wrap the configuration
+	// items so that the dependencies properly map to the modules.
+	// See the docs for graphNodeModuleWrappable for more info.
```

Comment:
is `graphNodeModuleWrappable` from an earlier iteration? do you mean `graphNodeModuleSkippable` now?